src/test/java/com/github/paweladamski/httpclientmock/HttpClientResponseBuilderTest.java

```diff
@@ -164,6 +167,21 @@ public void should_add_header_to_response() throws IOException {
         assertThat(second.getFirstHeader("tracking").getValue(), equalTo("456"));
     }
 
+    @Test
+    public void should_add_cookie_to_response() throws IOException {
+        HttpClientMock httpClientMock = new HttpClientMock("http://localhost:8080");
+        httpClientMock.onPost("/login")
+                .doReturn("foo").withCookie("cookieName", "cookieValue")
+                .doReturn("foo").withCookie("cookieName", "cookieValue2");
+
+        HttpClientContext httpClientContext = new HttpClientContext();
+        httpClientMock.execute(httpPost("http://localhost:8080/login"), httpClientContext);
+        assertThat(getCookieValue(httpClientContext.getCookieStore(),"cookieName"), equalTo("cookieValue"));
+
+        httpClientMock.execute(httpPost("http://localhost:8080/login"), httpClientContext);
+        assertThat(getCookieValue(httpClientContext.getCookieStore(),"cookieName"), equalTo("cookieValue2"));
```

Comment:
Would be nice if we you could  create a matcher similar to `HttpResponseMatchers#hasStatus` so we could write 
`assertThat(httpClientContext, hasCookie("cookieName", "cookieValue2"));

Reply:
Ok, I have created a `matcher`